dpdk/vr_dpdk_virtio.c

```diff
@@ -798,12 +814,16 @@ vr_dpdk_virtio_enq_pkts_to_phys_lcore(struct vr_dpdk_queue *rx_queue,
 
     vq = (vr_dpdk_virtioq_t *) rx_queue->q_queue_h;
     vq_pring = vq->vdv_pring;
+    if (!vq_pring)
+        return;
 
     RTE_LOG(DEBUG, VROUTER, "%s: enqueue %u pakets to ring %p\n",
                 __func__, npkts, vq_pring);
     nb_enq = rte_ring_sp_enqueue_burst(vq_pring, (void **) pkt_arr, npkts);
-    for ( ; nb_enq < npkts; nb_enq++)
-        vr_pfree(pkt_arr[nb_enq], VP_DROP_INTERFACE_DROP);
+
+    for ( ; nb_enq < npkts; nb_enq++) {
+        vif_drop_pkt(pkt_arr[nb_enq]->vp_if, pkt_arr[nb_enq], 0, VP_DROP_ENQUEUE_FAIL);
```

Comment:
Can we use a different error code for this enqueue (which is for TX from the VM) and the enqueue failure when we enqueue to a ring for packets being sent to the VM?